EmailValidator/EmailLexer.php

```diff
@@ -77,10 +77,16 @@ class EmailLexer extends AbstractLexer
 
     protected $previous;
 
+    public function __construct()
+    {
+        $this->reset();
+    }
+
     public function reset()
     {
         $this->hasInvalidTokens = false;
         parent::reset();
+        $this->previous = $this->token = ['type' => null, 'value' => '', 'position' => 0];
```

Comment:
I'd suggest to use `EmailLexer::C_NUL` for `type` instead.

Reply:
this wouldn't be the same - we don't want to this be confused with the NUL char.

Reply:
Agree. Should a new token be added?

Reply:
I don't think so, `null` is really the best "const" for this to me.